[AUTO plugin] AUTO plugin will ignore other plugins' configuration

mashoujiang opened the pull request:
### Details:
- *Auto plugin only support its config "IE::KEY_AUTO_DEVICE_LIST", no more*
- *Auto will use the config from LoadNetwork("AUTO", config) to extract the "IE::KEY_AUTO_DEVICE_LIST", but for other devices like CPU, AUTO will ignore the configuration, and will be "executableNetwork = GetCore()->LoadNetwork(param, selectedDevice, {});"*
- *If User need set those devices config, they should set by "ie.SetConfig(config, CPU);" *

### Tickets:
 - *57233*

myshevts commented on inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/config.cpp:
```diff
@@ -42,18 +42,7 @@ namespace {
 
     const std::vector<std::map<std::string, std::string>> AutoConfigs = {
-            {{InferenceEngine::KEY_AUTO_DEVICE_LIST , CommonTestUtils::DEVICE_CPU},
-                    {InferenceEngine::PluginConfigParams::KEY_CPU_THROUGHPUT_STREAMS, InferenceEngine::PluginConfigParams::CPU_THROUGHPUT_AUTO}},
```
maybe it was better to keep the test cases, but implement the variant of the InferConfigTests that sets the options via the ie.Setconfig directly to the plugin specified in the KEY_AUTO_DEVICE_LIST, as opposite to passing them to the LoadNetwork

mashoujiang replied:
Thanks Maxim, I didn't delete these tests but only remove CPU related config for AUTO plugin. and they have no difference by ie.SetConfig(config) or LoadNetwork(config) since both will ignore thoese config for CPU.

myshevts replied:
I was referring to setting config with ie.SetConfig(config, CPU) directly to the device, and then testing that AUTO works

mashoujiang replied:
The tests are in plugin/shared folder, so it may be used by other plugin. I think we could not add ie.SetConfig(config, CPU) to those common test code.

myshevts commented on inference-engine/tests/functional/plugin/cpu/shared_tests_instances/behavior/infer_request_config.cpp:
```diff
@@ -51,20 +51,7 @@ namespace {
     };
 
     const std::vector<std::map<std::string, std::string>> AutoConfigs = {
-            {{InferenceEngine::KEY_AUTO_DEVICE_LIST , CommonTestUtils::DEVICE_CPU},
```
again, maybe it was possible to leave the options, but implement a variant of the InferConfigTests that sets them directly to the plugin (via ie.SetConfig)

mashoujiang replied:
reply the same above

myshevts commented:
I have added a comment to the CVS-57233 that the AUTO should only ignore "AUTO_" config keys, and throw an exception for the rest

mashoujiang commented:
> I have added a comment to the CVS-57233 that the AUTO should only ignore "AUTO_" config keys, and throw an exception for the rest

Got it, I have updated the code. We will skip those tests which is not very special to AUTO and has hard code configurations like `EXCLUSIVE_ASYNC_REQUESTS` and `KEY_ENFORCE_BF16`

mashoujiang commented:
@myshevts @ilya-lavrenov could you please help to review again? all CI tests are passed now, thank you very much!

myshevts commented on inference-engine/tests/functional/plugin/shared/include/behavior/config.hpp:
```diff
@@ -73,8 +72,12 @@ namespace BehaviorTestsDefinitions {
         InferenceEngine::CNNNetwork cnnNet(function);
-        ASSERT_THROW(auto execNet = ie->LoadNetwork(cnnNet, targetDevice, configuration),
-                     InferenceEngine::Exception);
+        if (targetDevice.find(CommonTestUtils::DEVICE_AUTO) != std::string::npos) {
```
Why we skip a test for the AUTO, it should throw just like any other plugin

mashoujiang replied:
This test is modified in a early commit which has no exception, but I add the exception for unsupported config as CVS-57233. I created ticket #CVS-57552 to track this.

myshevts commented on inference-engine/src/auto_plugin/auto_plugin.cpp:
```diff
@@ -215,23 +209,37 @@ ConfigType AutoInferencePlugin::GetSupportedConfig(const ConfigType&  config,
+    for (auto&& c : config) {
+        auto itKey = std::find(supportedConfigKeys.begin(), supportedConfigKeys.end(), c.first);
+        if (supportedConfigKeys.end() == itKey) {
+            // CVS-57233
```
this is CVS-57238, not 57223

mashoujiang replied:
it should be https://jira.devtools.intel.com/browse/CVS-57233, right?

ilya-lavrenov commented on inference-engine/tests/functional/plugin/shared/include/behavior/infer_request_config.hpp:
```diff
@@ -71,8 +73,10 @@ TEST_P(InferConfigTests, withoutExclusiveAsyncRequests) {
-    execNet.CreateInferRequest();
+    if (targetDevice.find(CommonTestUtils::DEVICE_AUTO) == std::string::npos) {
+        auto execNet = ie->LoadNetwork(cnnNet, targetDevice, config);
+        execNet.CreateInferRequest();
```
@myshevts we must not have any ifs inside beh tests

mashoujiang replied:
yes, #6113 is fixing this issue.

myshevts replied:
@ilya-lavrenov , yes, I already pointed to that as well:
https://github.com/openvinotoolkit/openvino/pull/5979#discussion_r647231092

so this PR was merged to meet the FF deadline, and associated ticket to fix the tests was created
